wallet/src/account/mod.rs

```diff
@@ -658,6 +671,42 @@ impl Account {
         Ok(())
     }
 
+    pub fn scan_new_unconfirmed_transactions(
+        &mut self,
+        transactions: &[SignedTransaction],
+        db_tx: &mut impl WalletStorageWriteLocked,
+    ) -> WalletResult<()> {
+        let mut not_added = vec![];
+        for signed_tx in transactions {
+            let wallet_tx = WalletTx::Tx(TxData::new(
+                signed_tx.transaction().clone().into(),
+                TxState::InMempool,
+            ));
+
+            // in the case when 2 unconfirmed txs depend on each other, and the last one spends
+            // utxos that belong to this account (but has no outputs associated with this account)
+            // it is not possible to determine if that tx is for this account or not
+            // if we haven't processed the previous tx before it.
+            // This can only happen for the last one in the chain, as any other tx will have an
+            // output belonging to this account.
+            if !self.add_wallet_tx_if_relevant(db_tx, wallet_tx)? {
+                not_added.push(signed_tx);
+            }
+        }
+
+        // check them again after adding all we could
+        for signed_tx in not_added {
+            let wallet_tx = WalletTx::Tx(TxData::new(
+                signed_tx.transaction().clone().into(),
+                TxState::InMempool,
+            ));
+
+            self.add_wallet_tx_if_relevant(db_tx, wallet_tx)?;
+        }
```

Comment:
From my understanding no, as no matter the chain length in every transaction of the chain up until the last one you will always have an output that belongs to this wallet which will be identified in the first for loop. The last one that doesn't have an output but only consumes an input will be caught in the second one.
If a long chain has transactions that don't have neither an output nor an input belonging to the wallet, then you can think of that chain as multiple chains from the parts that we care about.

Reply:
I think there should be a loop. It is possible that the wallet does not recognize some transactions as its own because the address pool is not big enough (for example, if you use two wallets with the same mnemonic and the remote wallet tries to scan mempool).

Reply:
fixed